### PR TITLE
Sound Effect For Emagged Borgs

### PR DESCRIPTION
**Pull request by Bixkitts**
## About the PR
Added a sound effect that plays for the client when they get emagged as a cyborg and their laws change.

## Why / Balance
Because it's cool, and adds an audio queue to the client that they got got with an emag (feature request and sound by @ps3moira)

## Technical details
Added in a function call to play a sound globally for the client when emagged as a borg, similar to the sound queues that play when a traitor role is assigned.

## Media

- [ X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

- add: Added sound queue for borg players getting emagged

### CODE REVIEW

**Comment by deltanedas** [comment]
really need a generic thing to play sound when a role is assigned this is getting out of hand

**Comment by Bixkitts** [comment]
Shall I move this to MindAddRole() and make it generic?

**Comment by Bixkitts** [comment]
Well I'd say please merge or close this PR and either way I'll try make a generic thing in the next day or two.

**Comment by deltanedas** [comment]
i was thinking more a role component that subscribes to the mind-raised mind added event (if it exists) and plays a sound, ill also try to make something

**Comment by Bixkitts** [comment]
Yep there's a MindRoleAddedEvent, but I don't think it will be of use.
Each role (seemingly) has it's own component already and you could probably set the sound file attributed to it in those.
imo just grab the sound out of the "component" argument in MindAddRole<T>() (or rather, grab it from that component on the entity after ensuring the role was assigned),
grab the session from the mindId and play the sound with it.

**Comment by deltanedas** [comment]
#21460 :trollface:

**Comment by Emisse** [comment]
This sounds really good but yes needs deltas PR

File: Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs [hunk 1]
```diff
@@ -48,4 +49,12 @@ public sealed partial class EmagSiliconLawComponent : Component
     /// </summary>
     [DataField]
     public ProtoId<AntagPrototype>? AntagonistRole = "SubvertedSilicon";
+
+    /// <summary>
+    /// The sound that plays for the borg player
+    /// to let them know they've been emagged
+    /// </summary>
+    [DataField("emaggedSound")]
```
Comment (TheShuEd) [inline Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs:57]: just `[DataField]`  name in brackets is an obsolete format

File: Resources/Audio/Ambience/Antag/attributions.yml [hunk 2]
```diff
@@ -14,3 +14,7 @@
   license: "CC0-1.0"
   copyright: "Made by https://www.youtube.com/@a-guy173"
   source: https://github.com/A-Guy173/Music/blob/main/revolution.mp3
+- files: ["emagged_borg.ogg"]
+  license: "CC0-1.0"
+  copyright: "Made by @ps3moira on github"
+  source: https://www.youtube.com/watch?v=4-R-_DiqiLo
```
Comment (TheShuEd) [inline Resources/Audio/Ambience/Antag/attributions.yml:20]: I'm not sure if youtube videos count as good primary sources. There is no indication anywhere of the license used. I think this sound can be found somewhere either in ss13 builds, and make a more correct reference to the license and source.

Comment (Bixkitts) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: I made this as part of a request, allegedly it's self made and I was just given the video
and rolled with it.
Youtube videos have a date, user accounts, and are owned by a third party, so if it's got
the earliest time and date stamp to show it's the source it should be as legit as a reference 
to an SS13 repo?
The requester didn't request a specific license so I copy pasted creative commons.
Either way I'll follow up on all that.

File: Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs (hunk 1 above)
Comment (Bixkitts) [reply Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs:57]: This is what I've seen all over the codebase and had merged myself before, but I'll take note of that and change it.

Comment (TheShuEd) [reply Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs:57]: it seems that before they were mandatory, but now if they are not specified, then the same variable name will work in prototypes, only with a small letter.

Comment (Bixkitts) [reply Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs:57]: Handy!

File: Resources/Audio/Ambience/Antag/attributions.yml (hunk 2 above)
Comment (deltanedas) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: default license is not public domain its all rights reserved

Comment (Bixkitts) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: I literally just copy pasted the next line closest to my cursor at that moment,
I'm not a lawyer and didn't care and neither did the creator seemingly
but you're right, sane defaults and all that.
I'll watch out!

Comment (TheShuEd) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: I think you should publish the sound on freesound.com, specify the license there, and leave a link to this site here.

Comment (Bixkitts) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: Sure, I'll see to it.

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs [hunk 3]
```diff
@@ -212,6 +213,7 @@ private void EnsureEmaggedRole(EntityUid uid, EmagSiliconLawComponent component)
             return;
 
         _roles.MindAddRole(mindId, new SubvertedSiliconRoleComponent { PrototypeId = component.AntagonistRole });
+        _roles.MindPlaySound(mindId, component.EmaggedSound);
```
Comment (EmoGarbage404) [inline Content.Server/Silicons/Laws/SiliconLawSystem.cs:216]: this should probably just be in the code for getting emagged. This sound effect is like 20 seconds long and could be easily spammed by taking an MMI in and out of a chassis.

Comment (Bixkitts) [reply Content.Server/Silicons/Laws/SiliconLawSystem.cs:216]: Good observation, despite coding for it I haven't 
played around with borgs yet.
I'll look into it later and make sure it's unspammable.
THOUGH I _would_ like to get https://github.com/space-wizards/space-station-14/pull/21484 merged
and do it that way mayb.

File: Resources/Audio/Ambience/Antag/attributions.yml (hunk 2 above)
Comment (Bixkitts) [reply Resources/Audio/Ambience/Antag/attributions.yml:20]: Update on this, 
I consulted the creator about it.
They already have it uploaded, they can verify they're the creator, 
license info will be put in the youtube description.
I don't think it's 100% professional, and youtube not being open source
could theoretically be an issue?
Is that the reason?
idek, but seems good enough to me anyways.
If a reviewer takes issue with it, then we'll upload it somewhere else instead
as suggested!

**Comment by Bixkitts** [comment]
Moved it to trigger specifically on emag instead of EnsureEmaggedRole.

**Comment by ps3moira** [comment]
If someone is able to review the fixed changes for the PR, as it is all set up and ready, that'd be really appreciated!

**Comment by Bixkitts** [comment]
Tested in game, this seems to work fine.

**Pull request merged**